src/main/java/org/wise/portal/presentation/web/controllers/peergroup/PeerGroupAnnotationsAPIController.java

```diff
@@ -0,0 +1,79 @@
+
+import org.json.JSONException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.wise.portal.domain.peergroup.PeerGroup;
+import org.wise.portal.domain.peergroup.impl.PeerGroupImpl;
+import org.wise.portal.domain.peergrouping.PeerGrouping;
+import org.wise.portal.domain.run.Run;
+import org.wise.portal.domain.run.impl.RunImpl;
+import org.wise.portal.domain.user.User;
+import org.wise.portal.domain.workgroup.Workgroup;
+import org.wise.portal.domain.workgroup.impl.WorkgroupImpl;
+import org.wise.portal.service.peergroup.PeerGroupCreationException;
+import org.wise.portal.service.peergrouping.PeerGroupingNotFoundException;
+import org.wise.portal.service.workgroup.WorkgroupService;
+import org.wise.vle.domain.annotation.wise5.Annotation;
+
+@RestController
+@Secured("ROLE_USER")
+@RequestMapping("/api/peer-group")
+public class PeerGroupAnnotationsAPIController extends AbstractPeerGroupAPIController {
+  @Autowired
+  private WorkgroupService workgroupService;
+
+  @GetMapping("/{peerGroupId}/{nodeId}/{componentId}/annotations")
+  List<Annotation> getPeerGroupAnnotations(@PathVariable("peerGroupId") PeerGroupImpl peerGroup,
+      @PathVariable String nodeId, @PathVariable String componentId, Authentication auth) {
+    if (isUserInPeerGroup(peerGroup, auth)) {
+      List<Workgroup> teacherWorkgroups = getTeacherWorkgroups(
+          peerGroup.getPeerGrouping().getRun());
+      return peerGroupService.getStudentAnnotations(peerGroup, nodeId, componentId,
+          teacherWorkgroups);
+    } else {
+      throw new AccessDeniedException("Not permitted");
+    }
+  }
+
+  @Secured("ROLE_TEACHER")
+  @GetMapping("/{runId}/{workgroupId}/{nodeId}/{componentId}/annotations")
+  List<Annotation> getPeerGroupAnnotations(@PathVariable("runId") RunImpl run,
+      @PathVariable("workgroupId") WorkgroupImpl workgroup, @PathVariable String nodeId,
+      @PathVariable String componentId, Authentication auth)
+      throws JSONException, PeerGroupingNotFoundException, PeerGroupCreationException {
+    User user = userService.retrieveUserByUsername(auth.getName());
+    if (runService.isAllowedToViewStudentWork(run, user)) {
+      PeerGrouping peerGrouping = peerGroupingService.getByComponent(run, nodeId, componentId);
+      PeerGroup peerGroup = peerGroupService.getPeerGroup(workgroup, peerGrouping);
+      List<Workgroup> teacherWorkgroups = getTeacherWorkgroups(run);
+      return peerGroupService.getStudentAnnotations(peerGroup, nodeId, componentId,
+          teacherWorkgroups);
```

Comment:
I was a bit confused when I read this code because the method is called ```getStudentAnnotations()``` and we're passing in teacherWorkgroups. When I follow the code calls, I see that the new ```AnnotationDao.getAnnotations()``` that takes in a list of teacher workgroups. This seems like a lot of extra work just to call ```AnnotationDao.getAnnotationsToWorkgroups()```.

Instead, I think it would be simpler if we can expose ```AnnotationDao.getAnnotationsToWorkgroups()``` via AnnotationService and call that with the right workgroup set (which includes the teacher workgroup). What do you think?